arrows/core/video_input_image_list.cxx

```diff
@@ -469,18 +524,51 @@ ::read_from_file( std::string const& filename )
       resolved_file = data_dir + line;
       if ( ! ST::FileExists( resolved_file ) )
       {
-        VITAL_THROW( kwiver::vital::
+        VITAL_THROW( kv::
           file_not_found_exception, line,
               "could not locate file relative to \"" + data_dir + "\"" );
       }
     }
 
     this->m_files.push_back( resolved_file );
   } // end while
+
+  if ( c_sort_by_time )
+  {
+    sort_by_time( this->m_files );
+  }
 }
 
 
-// ------------------------------------------------------------------
+// ----------------------------------------------------------------------------
+void
+video_input_image_list::priv
+::sort_by_time( std::vector< kv::path_t >& files )
+{
+  std::map< kv::time_usec_t, kv::path_t > m_sorted_files;
+
+  for ( auto file : this->m_files )
+  {
+    kv::metadata_sptr md = m_image_reader->load_metadata( file );
+
+    if ( !md || !md->timestamp().has_valid_time() )
+    {
+      VITAL_THROW( kv::invalid_file, file, "Could not load time" );
+    }
+
+    m_sorted_files[ md->timestamp().get_time_usec() ] = file;
+  }
+
+  files.clear();
+
+  for ( auto file : m_sorted_files )
```

Comment:
This really ought to use [`std::copy`](https://en.cppreference.com/w/cpp/algorithm/copy)...

Do you care at all about performance? If yes, it would probably be better to create a heap ([`std::push_heap`](https://en.cppreference.com/w/cpp/algorithm/push_heap)) of {time, index} and unwind that ([`std::pop_heap`](https://en.cppreference.com/w/cpp/algorithm/pop_heap)) to create the sorted list.

Reply:
So, for #1007, I actually did some benchmarking, and while a heap *is* faster for a sufficiently large input list (10k files), "sufficiently large" is unlikely in practice... and it turns out that just translating the vector of names to a vector of name+time pairs and then using `std::sort` is fastest in AFAICT all instances. (I also expect it helps to `std::move` the file names...)